readme.md

```diff
@@ -196,6 +196,31 @@ serialized Java `StackTraceElement` objects indicating the server-side stack tra
 future version of http-remoting may replace the stack trace mechanism with a more OS-independent API for relaying stack
 traces.
 
+#### Serialization of Optional and Nullable objects
+`@Nullable` or `Optional<?>` fields in complex types are serialized using the standard Jackson mechanism:
+- a present value is serialized as itself (in particular, without being wrapped in a JSON object representing the `Optional` object)
+- an absent value is serialized as a JSON `null`.
+For example, assume a Java type of the form
+```java
+public final class ComplexType {
+    private final Optional<ComplexType> nested;
+    private final Optional<String> string;
+}
+```
+, and an instance
+```java
+ComplexType value = new ComplexType(
+        Optional.of(
+                new ComplexType(
+                        Optional.<ComplexType>absent(),
+                        Optional.<String>absent(),
+        Optional.of("baz"));
+```
+The JSON-serialized representation of this object is:
+```json
+{"nested":{"nested":null,"string":null},"string":"baz"}
```

Comment:
I don't know.

On Wed, Aug 17, 2016 at 9:48 AM Mark Elliot <notifications@github.com>
wrote:

> In readme.md
> <https://github.com/palantir/http-remoting/pull/208#discussion_r75160792>:
>
> > +    private final Optional<ComplexType> nested;
> > +    private final Optional<String> string;
> > +}
> > +```
> > +, and an instance
> > +```java
> > +ComplexType value = new ComplexType(
> > +        Optional.of(
> > +                new ComplexType(
> > +                        Optional.<ComplexType>absent(),
> > +                        Optional.<String>absent(),
> > +        Optional.of("baz"));
> > +```
> > +The JSON-serialized representation of this object is:
> > +```json
> > +{"nested":{"nested":null,"string":null},"string":"baz"}
>
> Would it also be valid to omit the null properties entirely?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/http-remoting/pull/208/files/425d4b9852ade2eda3530bf29a152a6685345305#r75160792>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGOdwa_dQbSq0jS42r8Pf7TiF3NglsOYks5qgzsxgaJpZM4JmppJ>
> .
>